### PR TITLE
Fix man page to mention race condition

### PR DESCRIPTION
**Pull request by rhatdan**
Also modify man page to talk about podman-container-logs as well as podman-logs.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@edsantiago @TomSweeneyRedHat PTAL

File: docs/podman-logs.1.md [hunk 1]
```diff
@@ -1,9 +1,11 @@
-% podman-logs(1)
+% podman-container-logs(1)
 
 ## NAME
-podman\-logs - Fetch the logs of one or more containers
+podman\-container\-logs (podman\-logs) - Fetch the logs of one or more containers
```
Comment (edsantiago) [inline docs/podman-logs.1.md]: It makes me uncomfortable to have the filename be `podman-logs` but the doc string be `podman-container-logs`. Would it make sense to rename the file as long as you're making this change? And possibly, to conform with podman-info and podman-rm, keep `podman-logs` as an alternative?

In the process of reviewing this I've found a number of other problems with the man pages; I think we're going to need another CI script. (E.g. look at the NAME for podman-play.1.md and podman-generate.1.md).

Comment (rhatdan) [reply docs/podman-logs.1.md]: Yes we can fix that in a different PR.  I have only been fixing them as I see them. but we need someone to globally go through them all and do a rename of the docs files. And then to go and create the man page aliases.

File: docs/podman-logs.1.md [hunk 2]
```diff
@@ -15,7 +17,11 @@ any logs at the time you execute podman logs
 
 **--follow, -f**
 
-Follow log output.  Default is false
+Follow log output.  Default is false.
+
+Note: If you are following a container which is removed on exit
+`podman run --rm ...`, then there is a chance the the log file will be removed
+before `podman logs` reads the final content.
```
Comment (edsantiago) [inline docs/podman-logs.1.md]: This looks reasonable. I still find it confusing that removing a log file causes `tail` not to see final output -- my expectation is that `tail` keeps the file open, so unlink shouldn't matter -- but I'll chalk that up to Go weirdness. This note is helpful, thanks.

**Comment by TomSweeneyRedHat** [comment]
LGTM
push and search tests are failing which I can't imagine is due to this PR.

File: docs/podman-logs.1.md (hunk 1 above)
Comment (cevich) [reply docs/podman-logs.1.md]: >  to need another CI script. 

#2648 isn't merged yet, feel free to slap a ``/hold`` on it.  I'm happy to stick in whatever else you'd like to add.  Just let me know or add in a separate PR.  We can't never has too many CI scripts :grin:

Comment (edsantiago) [reply docs/podman-logs.1.md]: No, it's huge. Enormous. The man pages are a mess. I'm working on a PR right now to fix the low-hanging fruit, but will need serious help for the rest. There's no way to add a CI script right now, it's too broken. Feel free to merge #2648.

Comment (cevich) [reply docs/podman-logs.1.md]: oh okay, was hoping it was maybe juss a l'il one-liner.

Suggestion: if the problems are large / wide-spread, perhaps call in a docs person to help?  No need to take it all on yourself :smiley:

**Comment by cevich** [comment]
```[Fail] Podman push [It] podman push to local registry with authorization```

***Yes*** I know about that, it's breaking everywhere, no idea why.  :frowning:

File: docs/podman-logs.1.md [hunk 3]
```diff
@@ -15,7 +17,11 @@ any logs at the time you execute podman logs
 
 **--follow, -f**
 
-Follow log output.  Default is false
+Follow log output.  Default is false.
+
+Note: If you are following a container which is removed on exit
+`podman run --rm ...`, then there is a chance the the log file will be removed
```
Comment (mheon) [inline docs/podman-logs.1.md]: I don't think this is specific to `--rm` - I imagine you can trigger this with `rm -f` on any running container. Maybe we should just say "following a container which is removed"?

File: docs/podman-logs.1.md (hunk 2 above)
Comment (giuseppe) [reply docs/podman-logs.1.md]: If we reopen the file on an inotify event then that is a bug

Comment (mheon) [reply docs/podman-logs.1.md]: We aren't using inotify, even, just polling. There's a bug in the Go inotify library that causes it to eat delete events.

**Pull request merged**